engine/src/test/java17/org/hibernate/validator/test/constraints/records/RecordConstrainedTest.java

```diff
@@ -0,0 +1,240 @@
+	}
+
+	@Test
+	public void testExplicitConstructorRecord() {
+		try {
+			new ConstructorValidationRecord( "David", 15 );
+		}
+		catch (ConstraintViolationException e) {
+			Assert.fail( "No violations expected" );
+		}
+
+		try {
+			new ConstructorValidationRecord( null, 15 );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( NotBlank.class ).withMessage( "Name cannot be null or empty" ) );
+		}
+
+		try {
+			new ConstructorValidationRecord( "", 15 );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( NotBlank.class ).withMessage( "Name cannot be null or empty" ) );
+		}
+
+		try {
+			new ConstructorValidationRecord( " ", 15 );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( NotBlank.class ).withMessage( "Name cannot be null or empty" ) );
+		}
+
+		try {
+			new ConstructorValidationRecord( "David", 0 );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( Positive.class ).withMessage( "Age has to be a strictly positive integer" ) );
+		}
+
+		try {
+			new ConstructorValidationRecord( "David", -15 );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( Positive.class ).withMessage( "Age has to be a strictly positive integer" ) );
+		}
+	}
+
+	@Test
+	public void testCompactConstructorRecord() {
+		try {
+			new CompactConstructorValidationRecord( "David" );
+		}
+		catch (ConstraintViolationException e) {
+			Assert.fail( "No violations expected" );
+		}
+
+		try {
+			new CompactConstructorValidationRecord( null );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( NotBlank.class ).withMessage( "Name cannot be null or empty" ) );
+		}
+
+		try {
+			new CompactConstructorValidationRecord( "" );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( NotBlank.class ).withMessage( "Name cannot be null or empty" ) );
+		}
+
+		try {
+			new CompactConstructorValidationRecord( " " );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( NotBlank.class ).withMessage( "Name cannot be null or empty" ) );
+		}
+	}
+
+	@Test
+	public void testRecordMethodValidation() {
+		try {
+			new MethodValidationRecord( 50 );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( Positive.class ) );
+		}
+	}
+
+	@Test
+	public void byteBuddyRecordTest() {
```

Comment:
I added this test, even if I didn't actually manage to get it to work; perhaps I'm not adding the validation annotation to the record class correctly, or in order for this 'hot swap' to work we would need a java agent?